doc/sphinx/proofs/writing-proofs/proof-mode.rst

```diff
@@ -738,6 +738,88 @@ When a focused goal is proved, Rocq displays a message suggesting use of
    - "None": this makes bullets inactive.
    - "Strict Subproofs": this makes bullets active (this is the default behavior).
 
+.. _named_goals:
+
+Named goals
+~~~~~~~~~~~
+
+You can focus on a goal by using its name. Goals do not have a name by default,
+but a name can be given by using :n:`refine ?[@ident]`, or generated using the
+:flag:`Generate Goal Names` flag.
+
+.. flag:: Generate Goal Names
+
+   Enable automatic generation of goal names for the :tacn:`induction`,
+   :tacn:`destruct` and :tacn:`eapply` tactics. For :tacn:`induction` and
+   :tacn:`destruct`, the subgoal takes the name of the corresponding
+   constructor. For :tacn:`eapply`, the subgoal takes the name of the
+   corresponding hypothesis.
+
+   This option makes it possible to write proofs with multiple subgoals that do
+   not depend on the order in which constructors were defined, but instead rely
+   on the constructor names. If you use bullets or numbers, reordering
+   constructors will break the proof.
+
+   For proofs that use nested :tacn:`induction` or case analysis, qualified
+   names such as `true.false` are used to disambiguate subgoals (see an example
+   :ref:`here <qualified-goal-names>`).
+
+   .. example:: Automatic generation of goal names
+
+      For the example given :ref:`here <example-working-with-named-goals>`,
+      names are generated for both the base case and the induction case.
+
+      .. rocqtop:: all
+
+         Set Generate Goal Names.
+
+         Goal forall n, n + 0 = n.
+         induction n.
+         [O]: { (* O is the name of the constructor for zero. *)
+
+      .. rocqtop:: in abort
+
+           reflexivity.
+         }
+
+      This flag also gives a name to goals that come from a binder or hypothesis:
+
+      .. rocqtop:: all reset
+
+         Set Generate Goal Names.
+
+         Goal exists n : nat, n = n.
+         eexists.
+         [n]: exact 0.
+         reflexivity.
+         Qed.
+
+   .. _qualified-goal-names:
+   .. example:: Qualified goal names
+
+      When doing nested case analysis or induction, qualified names are used to
+      disambiguate subgoals. The name of each subgoal is prefixed by the name of
+      its parent.
+
+      .. rocqtop:: all
```

Comment:
```suggestion
      .. rocqtop:: in
```

This example becomes almost two screenfuls (tedious for the user to scan to figure out what parts are relevant)--much more understandable if you don't show all the output.  Leave it as an exercise for the reader to try the example.